core/src/com/projectkorra/projectkorra/ability/WaterAbility.java

```diff
@@ -282,6 +286,35 @@ public static boolean isCauldron(final Material material) {
 		return GeneralMethods.getMCVersion() >= 1170 && (material == Material.getMaterial("WATER_CAULDRON") || material == Material.getMaterial("POWDER_SNOW_CAULDRON"));
 	}
 
+	public static boolean isSponge(final Block block) {
+		return block != null ? isSponge(block.getType()) : false;
+	}
+
+	public static boolean isSponge(final Material material) {
+		return material == Material.WET_SPONGE;
+	}
+
+	// A "non-transparent source" is a PK supported default non-transparent source block, not one from the config.
+	// I.e, cauldrons, mud and sponges.
+	// Used so that non-transparent sources aren't replaced with AIR.
+	public static boolean isNonTransparentSource(final Block block) {
+		return isNonTransparentSource(block.getType());
+	}
+
+	public static boolean isNonTransparentSource(final Material material) {
+		return isCauldron(material) || isMud(material) || isSponge(material);
+	}
+
+	// A "solid source" is a PK supported default solid source block, not one from the config.
+	// I.e., cauldrons, mud, sponges, ice and solid snow.
+	public static boolean isSolidSource(final Block block) {
+		return isSolidSource(block.getType());
+	}
```

Comment:
Should switch to actual javadoc formatting here

Reply:
Agreed, was just a quick comment lol.

Reply:
- Addressed in commit https://github.com/ProjectKorra/ProjectKorra/pull/1329/commits/5c4dd4c8c9e00a4c246d5f9009d3878ebea5e6bd.